assets/chat/js/formatters/UrlFormatter.js

```diff
@@ -70,9 +70,23 @@ export default class UrlFormatter {
       const decodedUrl = self.elem.html(url).text();
       const m = decodedUrl.match(self.linkregex);
       if (m) {
-        const encodedUrl = self.encodeUrl(m[0]);
+        let encodedUrl = self.encodeUrl(m[0]);
+
         const extra = self.encodeUrl(decodedUrl.substring(m[0].length));
         const href = `${scheme ? '' : 'http://'}${encodedUrl}`;
+
+        const maxUrlLength = 90;
+
+        if (encodedUrl.length > maxUrlLength) {
+          const midpoint = Math.ceil(encodedUrl.length / 2);
+          const toRemove = encodedUrl.length - maxUrlLength;
+          const strip = Math.ceil(toRemove / 2);
+          encodedUrl = `${encodedUrl.substring(
+            0,
+            midpoint - strip
+          )}...${encodedUrl.substring(midpoint + strip)}`;
```

Comment:
It's a little difficult to follow what's going on here. I don't think it's necessary to calculate the number of characters that need to be removed.

Can you just take the first `maxUrlLength / 2` and the last `maxUrlLength / 2` of the URL and concatenate with an ellipsis in between? You can use [`slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) to easily extract a substring from the end.

Reply:
I just realized a URL that only just exceeds the `maxUrlLength` will actually become a little longer with the ellipsis. Maybe we should set the limit at 90 but shorten to 80 to combat this.